test/services/ldap/strategies/univention.test.js

```diff
@@ -0,0 +1,222 @@
+const { expect } = require('chai');
+const sinon = require('sinon');
+
+const appPromise = require('../../../../src/app');
+
+const AbstractLDAPStrategy = require('../../../../src/services/ldap/strategies/interface');
+const UniventionLDAPStrategy = require('../../../../src/services/ldap/strategies/univention');
+
+const mockLDAPConfig = {
+	url: 'ldaps://foo.bar:636',
+	providerOptions: {},
+};
+
+describe('UniventionLDAPStrategy', () => {
+	it('implements AbstractLDAPStrategy', () => {
+		expect(new UniventionLDAPStrategy()).to.be.instanceOf(AbstractLDAPStrategy);
+	});
+
+	let originalLdapService;
+	let ldapServiceMock;
+
+	let app;
+
+	before(async () => {
+		app = await appPromise;
+		originalLdapService = app.service('ldap');
+	});
+
+	after(() => {
+		app.use('/ldap', originalLdapService);
+	});
+
+	describe('#getSchools', () => {
+		function MockLdapService() {
+			return {
+				setup: () => {},
+				searchCollection: sinon.fake.returns(
+					Promise.resolve([
+						{ ou: '1', displayname: 'Testschule-1' },
+						{ ou: '2', displayname: 'Testschule-2' },
+					])
+				),
+			};
+		}
+
+		beforeEach(() => {
+			ldapServiceMock = new MockLdapService();
+			app.use('/ldap', ldapServiceMock);
+		});
+
+		it('should search at the given root path and fall back to empty string', async () => {
+			const configWithoutRootPath = {
+				url: 'ldaps://foo.bar:636',
+				providerOptions: {},
+			};
+			await new UniventionLDAPStrategy(app, configWithoutRootPath).getSchools();
+			expect(ldapServiceMock.searchCollection.calledWith(configWithoutRootPath, '')).to.equal(true);
+
+			const mockConfig = {
+				url: 'ldaps://foo.bar:636',
+				rootPath: 'cn=foo,cn=bar',
+				providerOptions: {},
+			};
+			await new UniventionLDAPStrategy(app, mockConfig).getSchools();
+			expect(ldapServiceMock.searchCollection.calledWith(mockConfig, mockConfig.rootPath)).to.equal(true);
```

Comment:
we are missing here 3rd param, options
It makes sense to include that filter in the string, in case someone wants to change it in code the results will be affected and the unit tests should make sure that is ok
```
const options = {
			filter: `(&(univentionObjectType=container/ou)(!(ucsschoolRole=school:ou:no_school))(objectClass=ucsschoolOrganizationalUnit)${ignoredSchools})`,
			scope: 'sub',
			attributes: [],
		};
		
```